### PR TITLE
Hide the 'don't save' button when in change mode. closes #247

### LINKED ISSUE #247: Hide the "Don't Save" button when discarding a file isn't a desired behavior
Currently, if a file is a brand new (unsaved) file, the Save/Save As prompt has a "Don't Save" button.  This is desired when a user attempts to change from a new file to a previously saved file and is prompted to save.  However, this is incorrect UI behavior when the user is just using the Save/Save As functionality of the editor.  We need to hide the "Don't Save" button when we are not in the "Changing File" mode.

### PR DESCRIPTION
**Pull request by urbantumbleweed**
#### What's this PR do?

It removes the "don't save" button from the save overlay when in change_file mode.
#### What are the important parts of the code?

In handlers the `showSaveOverlay` function now takes a boolean flag to hide/show button.  Also, in the overlays.js plugin there is a new case that passes the boolean as props.
#### How should this be tested by the reviewer?

`git pull origin remove-dont-save-button-on-change-file && npm i && npm run build`
Then launch the app and create new files and then either save or navigate away.
#### Is any other information necessary to understand this?

When saving a new file from button clicks or keyboard shortcut the 'don't save' button is removed.
When navigating to another file from a new file then 'don't save' button appears.
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

closes #247
#### Screenshots (if appropriate)

screen shot 2015-09-11 at 5 31 29 pm
screen shot 2015-09-11 at 5 31 18 pm

### CODE REVIEW

File: src/creators/show-save-overlay.js [hunk 1]
```diff
@@ -5,14 +5,16 @@ const {
 } = require('../constants/action-types');
 
 const {
-  SAVE_OVERLAY
+  SAVE_OVERLAY,
+  SAVE_ON_CHANGE_OVERLAY
 } = require('../constants/overlay-states');
 
-function showSaveOverlay(){
+function showSaveOverlay(showDontSaveButton){
```
Comment (phated) [inline src/creators/show-save-overlay.js]: magic boolean parameters aren't a good pattern.  Just make 2 creators

File: src/plugins/handlers.js [hunk 2]
```diff
@@ -225,8 +225,8 @@ function handlers(app, opts, done){
     store.dispatch(creators.showDeleteProjectOverlay());
   }
 
-  function showSaveOverlay(){
-    store.dispatch(creators.showSaveOverlay());
+  function showSaveOverlay(showDontSaveButton){
```
Comment (phated) [inline src/plugins/handlers.js]: magic boolean parameters aren't a good pattern. just make 2 handlers

File: src/views/save-overlay.js [hunk 3]
```diff
@@ -107,7 +107,7 @@ class SaveOverlay extends React.Component {
     const { filename } = this.state;
 
     let dontSaveButton;
-    if(isNew){
+    if(isNew && this.props.showDontSaveButton){
```
Comment (phated) [inline src/views/save-overlay.js]: destructure from props

**Comment by urbantumbleweed** [comment]
Ok, it works. Ready for review.

**Pull request merged**